crates/uv-distribution/src/metadata/lowering.rs

```diff
@@ -74,171 +72,224 @@ impl LoweredRequirement {
+                    (source, marker)
                 }
-                url_source(url, subdirectory.map(PathBuf::from))?
-            }
-            Source::Path { path, editable } => {
-                if matches!(requirement.version_or_url, Some(VersionOrUrl::Url(_))) {
-                    return Err(LoweringError::ConflictingUrls);
+                Source::Path {
+                    path,
+                    editable,
+                    marker,
+                } => {
+                    if matches!(requirement.version_or_url, Some(VersionOrUrl::Url(_))) {
+                        return Err(LoweringError::ConflictingUrls);
+                    }
+                    let source = path_source(
+                        PathBuf::from(path),
+                        Origin::Project,
+                        dir,
+                        dir,
+                        editable.unwrap_or(false),
+                    )?;
+                    let marker = marker.map(MarkerTree::from).unwrap_or_default();
+                    (source, marker)
                 }
-                path_source(
-                    PathBuf::from(path),
-                    Origin::Project,
-                    dir,
-                    dir,
-                    editable.unwrap_or(false),
-                )?
-            }
-            Source::Registry { index } => registry_source(&requirement, index)?,
-            Source::Workspace { .. } => {
-                return Err(LoweringError::WorkspaceMember);
-            }
-            Source::CatchAll { .. } => {
-                // Emit a dedicated error message, which is an improvement over Serde's default
-                // error.
-                return Err(LoweringError::InvalidEntry);
-            }
-        };
-        Ok(Self(Requirement {
-            name: requirement.name,
-            extras: requirement.extras,
-            marker: requirement.marker,
-            source,
-            origin: requirement.origin,
+                Source::Registry { index, marker } => {
+                    let source = registry_source(&requirement, index)?;
+                    let marker = marker.map(MarkerTree::from).unwrap_or_default();
+                    (source, marker)
+                }
+                Source::Workspace { .. } => {
+                    return Err(LoweringError::WorkspaceMember);
+                }
+                Source::CatchAll { .. } => {
+                    // Emit a dedicated error message, which is an improvement over Serde's default
+                    // error.
+                    return Err(LoweringError::InvalidEntry);
+                }
+            };
+
+            marker.and(requirement.marker.clone());
+
+            Ok(Self(Requirement {
+                name: requirement.name.clone(),
+                extras: requirement.extras.clone(),
+                marker,
+                source,
+                origin: requirement.origin.clone(),
+            }))
```

Comment:
This file is mostly just the code being indented and put under an iterator. Very minimal changes.